src/utils/common/common.c

```diff
@@ -1136,8 +1136,14 @@ metric_family_t *plugin_value_list_to_metric_family(value_list_t const *vl,
   fam->name = name.ptr;
   name = (strbuf_t){0};
 
-  fam->type =
-      (dsrc->type == DS_TYPE_GAUGE) ? METRIC_TYPE_GAUGE : METRIC_TYPE_COUNTER;
+  switch (dsrc->type) {
```

Comment:
You missed the breaks in switch cases. Without them you will go to the next case and your family type in all cases will be METRIC_TYPE_UNTYPED

Reply:
Thanks a lot, Barbara! I didn't know that switches in C work this way. I'll fix it